packages/server-wallet/src/__test__/create-channel/fake-funding.test.ts

```diff
@@ -81,46 +81,21 @@ it('Create a fake-funded channel between two wallets ', async () => {
     turnNum: 0,
   });
 
-  // B sends countersigned PreFund0 to A
+  // B sends countersigned PreFund0 to A _and_ PostFund03(since it is unfunded)
```

Comment:
Is this right? I think A hasn't signed `PostFund3` yet? Can we keep the fake funded case as close as possible to the direct funded case?

Reply:
I think it is right, yes.

> Can we keep the fake funded case as close as possible to the direct funded case?

This depends, why would we do that? Direct funding requires the channel to be funded, and so you need to call `updateFunding` on the wallet. Fake funding does not require the channel to be funded, so you can automatically sign the post fund setup without delay. This seems like the expected behaviour

Reply:
I think we might do so if fake funding is really a sort of developer tool that allows funding to be skipped, but otherwise makes the wallet behave as similarly as possible, with the idea that fake funded channels most likely get upgraded to funded channels with minimal fuss later on. I don't know of any other use case for a fake funded channel.

Reply:
I guess I don't understand what `Unfunded` means, then. Using `Direct` and calling `updateChannelFunding` without actually doing any funding, or running a blockchain at all, would be the same thing as what you're describing.

Reply:
I guess that is the current way we do "fake funding". I think we ought to deprecate `updateChannelFunding` entirely (eventually), since presumably the chain watcher will provide this information. Then, we can keep fake funding functionality by passing in "strategy:fake" and have the wallet auto-skip waiting for funding events. Which is exactly what you have here, up to the actual name you're using. And the other test in this dir can eventually have the `updateChannelFunding` calls removed when we are ready to switch to the chain watcher?

Reply:
> since presumably the chain watcher will provide this information

The chain watcher will / already provides the information by calling `updateChannelFunding`, right @kerzhner ?

Reply:
The chain service does *not* use `updateChanneFunding` at the moment. When the wallet registers for chain events, the wallet passes in an object that implements the [`ChainEventSubscriberInterface`](https://github.com/statechannels/statechannels/blob/e7f4098be07fce36cc6f8115f25e32d4d1668a80/packages/server-wallet/src/chain-service/chain-service.ts#L42). Currently, the wallet implements that interface. The chain service will then call `onHoldingUpdated` method on the object when a `Deposit` event should be relayed to the wallet. `onHoldingUpdated` has a slightly different signature than `updateChannelFunding`.

Reply:
Though it looks like `onHoldingUpdated` calls `updateChannelFunding` (incorrectly) at the moment: https://github.com/statechannels/statechannels/blob/master/packages/server-wallet/src/wallet/index.ts#L519